runtime/vm/CRIUHelpers.cpp

```diff
@@ -135,6 +135,34 @@ isNonPortableRestoreMode(J9VMThread *currentThread)
 	return currentThread->javaVM->checkpointState.isNonPortableRestoreMode;
 }
 
+BOOLEAN
+isJdwpDebugThread(J9VMThread *thread, const char *threadName)
+{
+	bool result = FALSE;
+	j9object_t threadObject = thread->threadObject;
+	j9object_t threadGroup = NULL;
+	J9JavaVM *vm = thread->javaVM;
+
+	if (vm->checkpointState.isJdwpEnabled) {
+		if (NULL != threadObject) {
+#if JAVA_SPEC_VERSION >= 19
+			j9object_t threadHolder = J9VMJAVALANGTHREAD_HOLDER(thread, threadObject);
+			if (NULL != threadHolder) {
+				threadGroup = (j9object_t)J9VMJAVALANGTHREADFIELDHOLDER_GROUP(thread, threadHolder);
+			}
+#else /* JAVA_SPEC_VERSION >= 19 */
+			threadGroup = (j9object_t)J9VMJAVALANGTHREAD_GROUP(thread, threadObject);
+#endif /* JAVA_SPEC_VERSION >= 19 */
+		}
+		if (NULL != threadGroup && NULL != threadName) {
+			result = (threadGroup == *(vm->systemThreadGroupRef))
+					&& J9_ARE_ANY_BITS_SET(thread->privateFlags, J9_PRIVATE_FLAGS_DAEMON_THREAD)
+					&& (0 == strncmp("JDWP", threadName, 4));
```

Comment:
The JDWP agent does keep a list of the JDWP threads but I don't think the VM has access to that.

The additional checks should make it so there are no collisions with user created thread unless they specifically create a thread specifying the system threadgroup and as a daemon.

Reply:
It should not be possible for a user-created thread to be mistakenly considered a JDWP thread, even if we might agree it would be unusual user behavior.